ClientApp/KachnaOnline/src/app/shared/services/authentication.service.ts

```diff
@@ -417,8 +432,25 @@ export class AuthenticationService {
         this.storeUserDataToStorage();
         this.toastr.success("Změna souhlasu s gamifikací proběhla úspěšně.", "Změna uživatelských údajů");
       }).catch((error: any) => {
-      throwError(error);
-      this.toastr.error("Změna souhlasu s gamifikací selhala.", "Změna uživatelských údajů");
+        throwError(error);
+        this.toastr.error("Změna souhlasu s gamifikací selhala.", "Změna uživatelských údajů");
+    });
+  }
+
+  changeCardCode(cardCode: string) {
+    this.http.put<KisLoggedInUserInformation>(`${environment.kisApiUrl}/users/me/rfid`,
+      JSON.stringify(cardCode.toUpperCase()), {headers: new HttpHeaders({'Content-Type': 'application/json'})}).toPromise()
+      .then((res: KisLoggedInUserInformation) => {
+        this.kisLoggedInUserInformation = res;
+        localStorage.setItem(environment.kisLoggedInUserInformationStorageName, JSON.stringify(this.kisLoggedInUserInformation));
+
+        this.assignDataFromKisUserInformation();
+        this.assignDataFromLocalTokenContent();
+        this.storeUserDataToStorage();
+        this.toastr.success("Nastavení přiřazovacího kódu karty proběhlo úspěšně.", "Změna uživatelských údajů");
+      }).catch((error: any) => {
+        throwError(error);
+        this.toastr.success("Nastavení přiřazovacího kódu karty selhalo.", "Změna uživatelských údajů");
     });
```

Comment:
It would be great to provide a more specific error message for the different KIS response codes:
- 404: "Zadaný přiřazovací kód neexistuje, vyžádejte si u baru nový."
- 409: "Vaše karta nemůže být použita (v systému už existuje karta se stejným ID)."
- 429: "Karta může být změněna pouze jednou za XX dní."

Reply:
(I'm not sure what the timeout is, when I find it out, I'll update this comment.)

Reply:
Done. When the precise timeout is known, I will update the message.

Reply:
Done.